cmd/cli/plugin/package/main.go

```diff
@@ -52,3 +54,48 @@ func getOutputFormat() string {
 	}
 	return format
 }
+
+func isPackageMetadataAPIAvailable(kubeCfgPath string) (bool, error) {
+	cfg, err := kappclient.GetKubeConfig(kubeCfgPath)
+	if err != nil {
+		return false, err
+	}
+	clusterQueryClient, err := capdiscovery.NewClusterQueryClientForConfig(cfg)
+	if err != nil {
+		log.Error(err, "failed to create a new instance of the cluster query builder")
+		return false, err
+	}
+
+	apiGroup := capdiscovery.Group("packageMetadateAPIQuery", "data.packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packagemetadatas")
```

Comment:
I think you can combine all three queries into one with discovery client. cc @rajathagasthya

Reply:
Yep, you can combine by adding multiple arguments to `Query()`.

```go
q1 := capdiscovery.Group("packageMetadateAPIQuery", "data.packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packagemetadatas")
q2 := capdiscovery.Group("packageRepositoryAPIQuery", "packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packagerepositories")
q3 := capdiscovery.Group("packageInstallAPIQuery", "packaging.carvel.dev").WithVersions("v1alpha1").WithResource("packageinstalls")

clusterQueryClient.Query(q1, q2, q3).Execute()
```

Reply:
thanks for the review @vijaykatam , I'm wondering should we combine all the API checks together?

for example, `package available` only needs packageMetadateAPI check, `package repository` only needs packageRepositoryAPI check

Reply:
bring @maralavi into this discussion

Reply:
@ggpaue It won't hurt to check the availability of all apis for all commands. Anyways whoever is going to use package plugin needs all of those be available (eventhough when one of those available, almost always the remaining related apis for packaging are also available). I assume the discovery calls are quite fast and you could merge all these 3 into one query and use this combined approach as suggested above, and you could call that combined query for all commands.

Reply:
updated the code to put all API queries together